### ISSUE TITLE
ENH: speed up end beamtime

### ISSUE THREAD

**Issue by CJ-Wright** [xpdAcq/xpdAcq#398]
Would it be possible to rename the folder, run the compression in the background and then allow the next use to start?
Related we may want an end beamtime progress bar.

**Comment by CJ-Wright** [comment]
@sbanerjee23

**Comment by chiahaoliu** [comment]
Similar to #304. Rename dir is a better behavior and working in the background is desired. However, workflow gets complicated when user wants to interrupt ``_end_beamtime`` process that is in another thread (so that we need to recover all the directories.)

Proposed solution:

1. add directory rename step in to ``_end_beamtime``
2. require execute``_end_beamtime`` in a terminal different from the terminal that running the new beamtime.  (That's desirable since we would like to a fresh name space)

**Comment by CJ-Wright** [comment]
What is the recovery process in the event of an aborted end beamtime?

**Comment by CJ-Wright** [comment]
I'm not a huge fan of running it in a separate terminal. We may be able to separate the name spaces ourselves. I worry that people won't do it since they have another terminal to worry about.

**Comment by chiahaoliu** [comment]
User can't start a new beamtime unless he ends a beamtime so they won't forget to do it. And as you said, once a new beam time is started, the interest for old one would drop to almost zero. User might not even have time to think about the returned information of end_bemtime. We can use this fact to design a workflow. 

I can explain my reasoning more in person.

**Comment by CJ-Wright** [comment]
We might want to reassess the archival utility if the zipped tiffs and the zipped user folders in general.

**Comment by sbillinge** [comment]
This is extremely important psychologically for users.   However, any other
method that serves the same purpose is fine.

The original logic was to give users a clean work area, allow them to do
what they like in there, including exporting jpgs if they like, the
archiving it at the end so they know they can get it when they want it, and
in principle take it home with them in some kind of archive format too.

When we move to non-file-based system that the users understand and trust,
we can do away with this, but this won't be for a few years after we
introduce the non-file-based system and they learn to use and trust it
alongside their files.

Is this slowness of archiving still an issue?  Who is complaining about it?

**Comment by CJ-Wright** [comment]
@sbanerjee23 raised concerns about this at the most recent beamtime.

**Comment by sbillinge** [comment]
@sbanerjee, what is the concern?  That the software is not working?  If so,
how is it not working?

If the concern is that dark subtractions are not working, then I am not
interested in "concerns" about it but systematic investigation of the
problems and proposed solutions.  Can you prove to me that collecting a
dark every light produces a better PDF than using an occasional dark?

S

**Comment by chiahaoliu** [comment]
We are not currently compressing ``xpdUser`` at all, we just achieve and move ``xpdUser`` from local disk (workstation-2 or -3) to remote backup space.

Bottleneck comes from the network between local machine and remote  space ( I don't remember the number on top of my head, but it 's at the order of ~30MB/s or so). When user collected huge of amount of tiffs, issue could come up.

As per discussion with @CJ-Wright in person, potential solution could be the one list above or we exclude tiffs during achieving and provide a script/tool for beamline staff to recover all the tiffs that links to a given SAF number with the same file name.

**Comment by sbillinge** [comment]
Tim, please can we close this issue, and open an issue in it's place:

implement progress logger in end-beamtime.....if not too great of a
programming overhead.

Thanks.
S

**Comment by chiahaoliu** [comment]
Yes, of course. Created a new issue at #399

**Comment by chiahaoliu** [comment]
close and repost in #399
